### ISSUE TITLE
External I2C connection closes a ground loop

### ISSUE THREAD

**Issue by tomek-szczesny** [tomek-szczesny/OpenUPS#7]
Ah, snap. We may need to implement I2C isolator, or some passive hacks to prevent that from happening.

**Comment by tomek-szczesny** [comment]
On a related note, we can use unique AVR DB feature: A secondary voltage input for selected group of IO pins. It so happens that those pins include I2C Slave interface. Thus, we can use supply voltage coming from Qwiic for this group of pins.
This way, external I2C port will be compatible with any voltage levels up to 5V.

Hmm that doesn't help with the main issue here though.

**Comment by tomek-szczesny** [comment]
Oh yeah...
https://www.redeweb.com/en/articulos/aislamiento-sencillo-del-bus-i2c-mediante-componentes-discretos/?cn-reloaded=1

Guess we won't be needing that VCCIO2 feature after all.

**Comment by tomek-szczesny** [comment]
I found out that commercial, bidirectional I2C isolator (remember clock stretching) are generally unavailable and expensive AF.
I designed a optocoupler-based workaround that looks huge, power hungry and not that competitive in terms of price. Only when Vishay optocouplers are sourced from TME (3x cheaper for whatever reason!?), it barely makes any sense.
![2023-05-01-202057_1738x956_scrot](https://user-images.githubusercontent.com/44300715/235510008-fc9ccc0e-bf57-4840-b600-b18b89ab4026.png)
And then I found this chip, which is a direct replacement of AD and TI counterparts. 
![2023-05-01-203558_1681x860_scrot](https://user-images.githubusercontent.com/44300715/235510005-9c1c31b9-c71b-43b6-bc4b-8d4a9fd7e000.png)

Looks like the problem is solved.
